### PR TITLE
Add use_headings and slug options, fix image URLs conversion into local paths

### PR DESCRIPTION
**Pull request by artlomov**

### CODE REVIEW

File: foliant/backends/mkdocs.py [hunk 1]
```diff
@@ -15,7 +18,15 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         self._mkdocs_config = self.config.get('backend_config', {}).get('mkdocs', {})
-        self._mkdocs_site_dir_name = f'{self.get_slug()}.mkdocs'
+
+        overridden_slug = self._mkdocs_config.get('slug', None)
```
Comment (moigagoo) [inline foliant/backends/mkdocs.py]: The entire green section can be replaced with:

`self._mkdocs_site_dir_name = f'{self._mkdocs_config.get('slug', self.get_slug()}'`

File: foliant/backends/mkdocs.py [hunk 2]
```diff
@@ -44,6 +55,67 @@ def _get_ghp_command(self) -> str:
...
+        '''
+
+        def _sub(data_object, parent_is_dict):
+            if isinstance(data_object, dict):
+                new_data_object = {}
+                for key, value in data_object.items():
+                    new_data_object[key] = _sub(value, True)
+
+            elif isinstance(data_object, list):
+                new_data_object = []
```
Comment (moigagoo) [inline foliant/backends/mkdocs.py]: `data_object` is a bad variable name.

File: foliant/backends/mkdocs.py (hunk 1 above)
Comment (artlomov) [reply foliant/backends/mkdocs.py]: Replaced with:

`self._mkdocs_site_dir_name = f'{self._mkdocs_config.get("slug", self.get_slug())}.mkdocs'`

File: foliant/backends/mkdocs.py (hunk 2 above)
Comment (artlomov) [reply foliant/backends/mkdocs.py]: Replaced with `pages_component`.

**Pull request merged**